### ISSUE TITLE
OidcClient 3.1.2 is not compatible with IdentityModel 5.0.1

### ISSUE THREAD

**Issue by Flohack74** [DuendeArchive/IdentityModel.OidcClient#288]
We are currently getting this with the package combination mentioned above:
![grafik](https://user-images.githubusercontent.com/16186470/109781273-0fb47780-7c08-11eb-85e3-cea583ff2a7a.png)

It seems that OidcClient 3.1.2 is only compatible with IdentityModel up to 4.6 - can you confirm this? It would be good to have a restriction in place then.

**Comment by leastprivilege** [comment]
That's correct.

Well - generally speaking, you should go with the dependencies expressed by a package, unless you need a specific feature of a newer version. "Update all packages to latest" was never the right choice ;)

That being said - OidcClient 4.x is in preview and uses IdentityModel v5. Maybe that's an option.

If you think this is an issue that should be fixed, please open a PR. 

thanks!

**Comment by Flohack74** [comment]
I did not intentionally upgrade to 5.0.1, it just happened on a clean build machine that for whatever reason paket restored to 5.0.1 - for the time being we will pin it, but I was wondering if there shouldn´t be a restriction in the OidcClient NuGet package to not allow that resolution, if its a breaking change.
A preview version is not an option for us, though its for sure safe to use, we will remove the pin after 4.0 was released. Thanks!
